app/src/main/java/com/chess/clock/fragments/TimeSettingsFragment.java

```diff
@@ -357,8 +358,19 @@ public boolean onActionItemClicked(ActionMode actionMode, MenuItem menuItem) {
         if (menuItem.getItemId() == R.id.action_delete) {
             startBtn.performHapticFeedback(HapticFeedbackConstants.LONG_PRESS,
                     HapticFeedbackConstants.FLAG_IGNORE_GLOBAL_SETTING);
-            deleteTimeControls();
-            actionMode.finish(); // Action picked, so close the CAB
+            AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.ConfirmationDialogTheme);
+            builder
+                    .setMessage(R.string.delete_custom_time)
+                    .setPositiveButton(R.string.action_delete, (dialog, id) -> {
+                        deleteTimeControls();
+                        actionMode.finish(); // Action picked, so close the CAB
+                    })
+                    .setNegativeButton(R.string.action_keep, (dialog, id) -> {
+                        // Resume the clock
+                    });
+            Dialog dialog = builder.create();
+            ViewUtils.setUpConfirmationPopup(dialog, getResources());
+            dialog.show();
```

Comment:
Maybe if user didn't select anything to delete we should skip the dialog and show snackbar/Toast that one needs to select a time control first?

Reply:
Will add as separate task 👍